ansible_base/api_documentation/filter_extensions.py

```diff
@@ -0,0 +1,160 @@
+
+            if self._is_numeric_or_date_field(field_obj):
+                parameters.extend(self._create_comparison_parameters(field_name))
+
+        return parameters
+
+    def _get_field_by_name(self, model, field_name):
+        """Get field object by name from model."""
+        for field in model._meta.get_fields():
+            if hasattr(field, 'name') and field.name == field_name:
+                return field
+        return None
+
+    def _is_string_field(self, field):
+        """Check if field is a string-based field."""
+        from django.db import models
+
+        return isinstance(field, (models.CharField, models.TextField))
+
+    def _is_numeric_or_date_field(self, field):
+        """Check if field is numeric or date-based."""
+        from django.db import models
+
+        numeric_date_types = (models.IntegerField, models.DateTimeField, models.DateField, models.DecimalField, models.FloatField)
+        return isinstance(field, numeric_date_types)
+
+    def _create_comparison_parameters(self, field_name):
+        """Create comparison parameters (gt, gte, lt, lte) for a field."""
+        parameters = []
+        for lookup in ['gt', 'gte', 'lt', 'lte']:
+            parameters.append(self._create_parameter(f'{field_name}__{lookup}', f'Filter by {field_name} ({lookup})'))
+        return parameters
+
+    def _create_parameter(self, name, description):
+        """Create a single OpenAPI parameter."""
+        return build_parameter_type(
+            name=name,
+            schema={'type': 'string'},
+            location=OpenApiParameter.QUERY,
+            required=False,
+            description=description,
+        )
+
+    def _create_role_level_parameter(self):
+        """Create the role_level parameter for RBAC."""
+        return self._create_parameter('role_level', 'Filter by role level for RBAC')
+
+
+class TypeFilterBackendExtension(OpenApiFilterExtension):
+    target_class = TypeFilterBackend
+
+    def get_schema_operation_parameters(self, auto_schema, *args, **kwargs):
+        """
+        Generate OpenAPI parameters for TypeFilterBackend.
+
+        This filter backend supports filtering by object type.
+        """
+        return [
+            build_parameter_type(
+                name='type',
+                schema={'type': 'string'},
+                location=OpenApiParameter.QUERY,
+                required=False,
+                description='Filter by object type. Supports comma-separated values for multiple types.',
+            )
+        ]
+
+
+class OrderByBackendExtension(OpenApiFilterExtension):
+    target_class = OrderByBackend
+
+    def get_schema_operation_parameters(self, auto_schema, *args, **kwargs):
+        """
+        Generate OpenAPI parameters for OrderByBackend.
+
+        This filter backend supports ordering results by field names.
+        """
+        parameters = []
+
+        # Add the ordering parameters
+        for param_name in ['order', 'order_by']:
```

Comment:
I'm thinking order isn't appropriate here, but could be wrong.  In gateway at least, the models that do have an order field use it for other purposes (like hierarchical ordering, application/usage order, etc.).  order_by on the other hand is for ordering the query results.

Reply:
Ope, nevermind.  I see it isn't a field but a param and the order by backend supports both.  Ignore!